src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java

```diff
@@ -55,7 +55,13 @@ protected Void execute0(BuildImageCmd command, ResultCallback<BuildResponseItem>
         if (command.getTags() != null && !command.getTags().isEmpty()) {
             webTarget = webTarget.queryParamsSet("t", command.getTags());
         } else if (isNotBlank(command.getTag())) {
-            webTarget = webTarget.queryParam("t", command.getTags());
+            webTarget = webTarget.queryParam("t", command.getTag());
+        }
+
+        if (command.getCacheFrom() != null) {
+            for (String c: command.getCacheFrom()) {
+               webTarget = webTarget.queryParam("cachefrom", c);
```

Comment:
@KostyaSha You are right, there was confusion with 2 diff classes. The issue is fixed now as part of the new commit.